### ISSUE TITLE
CUDA.pool not defined

### ISSUE THREAD

**Issue by RocketRoss** [denizyuret/Knet.jl#595]
CUDA v1.2.1#master, Knet v1.3.9...

the gpufree() in gpu.jl:86 ```...CUDA.pool[].cached_memory()``` references an undefined value:

```
UndefVarError: pool not defined

Stacktrace:
 [1] gpufree() at C:\Users\Ross\.julia\packages\Knet\exwCE\src\gpu.jl:86
 [2] maxWorkspaceSize(::KnetArray{Float32,4}, ::KnetArray{Float32,4}, ::KnetArray{Float32,4}) at C:\Users\Ross\.julia\packages\Knet\exwCE\src\conv.jl:538
 [3] conv4_algo(::KnetArray{Float32,4}, ::KnetArray{Float32,4}, ::KnetArray{Float32,4}; handle::Ptr{Nothing}, o::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at C:\Users\Ross\.julia\packages\Knet\exwCE\src\conv.jl:550
 [4] conv4(::KnetArray{Float32,4}, ::KnetArray{Float32,4}; handle::Ptr{Nothing}, alpha::Int64, o::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at C:\Users\Ross\.julia\packages\Knet\exwCE\src\conv.jl:40
 [5] conv4(::KnetArray{Float32,4}, ::KnetArray{Float32,4}) at C:\Users\Ross\.julia\packages\Knet\exwCE\src\conv.jl:38
 [6] forw(::Function, ::AutoGrad.Param{KnetArray{Float32,4}}, ::Vararg{Any,N} where N; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at C:\Users\Ross\.julia\packages\AutoGrad\VFrAv\src\core.jl:66
 [7] forw at C:\Users\Ross\.julia\packages\AutoGrad\VFrAv\src\core.jl:65 [inlined]
 [8] #conv4#356 at .\none:0 [inlined]
 [9] conv4(::AutoGrad.Param{KnetArray{Float32,4}}, ::KnetArray{Float32,4}) at .\none:0
 [10] (::Conv)(::KnetArray{Float32,4}) at D:\Development\Code\Scripts\Julia\Notebooks\cnnStructs.jl:44
 [11] (::Chain)(::NTuple{4,Int64}; cast::Nothing) at D:\Development\Code\Scripts\Julia\Notebooks\cnnStructs.jl:164
 [12] (::Chain)(::NTuple{4,Int64}) at D:\Development\Code\Scripts\Julia\Notebooks\cnnStructs.jl:154
 [13] top-level scope at In[8]:2
```

**Comment by RocketRoss** [comment]
Could it be that it is intended to be ```CUDA.cached_memory()```? This is only available in CUDA#master, not CUDA v1.2.1.... This fixes the show-stopper on my pc.

I see that CUDA v1.2.0 is specced in the toml. Perhaps it that dependency should be restricted to 1.0.0-1.2.0. (:

**Comment by RocketRoss** [comment]
Confirmed that this issue does not exist with CUDA v1.2.0 and Knet v1.3.9#master...
@denizyuret
